### PR TITLE
Conditional compile improvements

### PR DESCRIPTION
**Pull request by vipera**
Improve conditional includes as outlined in #1334

### CODE REVIEW

File: lib/ngtcp2_unreachable.c [hunk 1]
```diff
@@ -35,6 +35,7 @@
 #endif /* WIN32 */
 
 void ngtcp2_unreachable_fail(const char *file, int line, const char *func) {
+#ifdef HAVE_UNISTD_H
```
Comment (tatsuhiro-t) [inline lib/ngtcp2_unreachable.c]: This breaks windows build.  See line62.

**Review by tatsuhiro-t — CHANGES_REQUESTED** [review]

File: lib/ngtcp2_unreachable.c (hunk 1 above)
Comment (vipera) [reply lib/ngtcp2_unreachable.c]: Thanks for pointing it out, I've added a common define to sort it out.

Comment (tatsuhiro-t) [reply lib/ngtcp2_unreachable.c]: Thank you. Looks good.

**Review by tatsuhiro-t — APPROVED** [review]

**Comment by tatsuhiro-t** [comment]
Thank you for PR.  Merged now.

**Comment by vipera** [comment]
Thanks for the clang-format as well ;)

**Pull request merged**